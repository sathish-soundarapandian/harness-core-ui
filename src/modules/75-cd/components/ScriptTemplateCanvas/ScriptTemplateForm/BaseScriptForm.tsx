/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { Formik, AllowedTypes } from '@wings-software/uicore'
import * as Yup from 'yup'
import type { FormikProps } from 'formik'

import { setFormikRef, StepFormikFowardRef } from '@pipeline/components/AbstractSteps/Step'
import { useStrings } from 'framework/strings'

import { ShellScriptFormData, variableSchema } from '@cd/components/PipelineSteps/ShellScriptStep/shellScriptTypes'
import BaseScript from '@cd/components/BaseScript/BaseScript'

interface ShellScriptWidgetProps {
  initialValues: ShellScriptFormData
  updateTemplate?: (data: ShellScriptFormData) => void
  onChange?: (data: ShellScriptFormData) => void
  allowableTypes: AllowedTypes
  readonly?: boolean
}

export function BaseScriptForm(
  { initialValues, updateTemplate, onChange, allowableTypes }: ShellScriptWidgetProps,
  formikRef: StepFormikFowardRef
): JSX.Element {
  const { getString } = useStrings()

  const defaultSSHSchema = Yup.object().shape({
    spec: Yup.object().shape({
      shell: Yup.string().trim().required(getString('validation.scriptTypeRequired')),
      source: Yup.object().shape({
        spec: Yup.object().shape({
          script: Yup.string().trim().required(getString('common.scriptRequired'))
        })
      }),
      environmentVariables: variableSchema(getString),
      outputVariables: variableSchema(getString)
    })
  })

  const values: any = {
    ...initialValues,
    spec: {
      ...initialValues.spec,

      executionTarget: {
        ...initialValues.spec?.executionTarget
      },
      onDelegate: initialValues.spec?.onDelegate ? 'delegate' : 'targethost'
    }
  }

  const validationSchema = defaultSSHSchema

  return (
    <Formik<ShellScriptFormData>
      onSubmit={/* istanbul ignore next */ submit => updateTemplate?.(submit)}
      validate={formValues => onChange?.(formValues)}
      formName="shellScriptForm"
      initialValues={values}
      validationSchema={validationSchema}
    >
      {(formik: FormikProps<ShellScriptFormData>) => {
        setFormikRef(formikRef, formik)

        return <BaseScript formik={formik} readonly={false} allowableTypes={allowableTypes} />
      }}
    </Formik>
  )
}

export const BaseScriptWithRef = React.forwardRef(BaseScriptForm)
