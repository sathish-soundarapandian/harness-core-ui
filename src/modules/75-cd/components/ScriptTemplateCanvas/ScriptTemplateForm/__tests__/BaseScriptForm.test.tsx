/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { MultiTypeInputType, Formik } from '@harness/uicore'
import { render, waitFor, act } from '@testing-library/react'
import { Form } from 'formik'
import { fillAtForm, InputTypes } from '@common/utils/JestFormHelper'
import { TestWrapper } from '@common/utils/testUtils'
import { BaseScriptWithRef } from '../BaseScriptForm'

const defaultInitialValues = {
  identifier: 'id',
  name: 'name',
  type: '',
  spec: {}
}
const defaultInitialValuesCorrect = {
  identifier: 'id',
  name: 'name',
  type: '',
  spec: {
    shell: 'Bash',
    source: {
      spec: {
        type: 'Inline',
        script: undefined
      }
    }
  }
}

const initialValues = {
  ...defaultInitialValues,
  spec: {
    shell: 'Bash',
    source: {
      spec: {
        type: 'Inline',
        script: 'echo test'
      }
    }
  }
}
const initialValuesWithExpressions = {
  ...defaultInitialValues,
  spec: {
    shell: 'Bash',
    source: {
      spec: {
        type: 'Inline',
        script: '<+spec.environmentVariable.var1>'
      }
    }
  }
}

const initialValuesWithRI = {
  ...defaultInitialValues,
  spec: {
    shell: 'Bash',
    source: {
      spec: {
        type: 'Inline',
        script: '<+input>'
      }
    }
  }
}

describe('Test BaseScriptWithRef', () => {
  test('initial render', async () => {
    const { container } = render(
      <TestWrapper>
        <BaseScriptWithRef initialValues={defaultInitialValues} allowableTypes={[]} />
      </TestWrapper>
    )

    expect(container).toMatchSnapshot()
  })
  test('should match snapshot for BaseScriptWithRef with initial values', async () => {
    const { container, getByText } = render(
      <TestWrapper>
        <BaseScriptWithRef initialValues={initialValues} allowableTypes={[MultiTypeInputType.EXPRESSION]} />
      </TestWrapper>
    )
    await waitFor(() => getByText('echo test'))
    expect(container).toMatchSnapshot()
  })

  test('should match snapshot for BaseScriptWithRef with initial script values with expressions', async () => {
    const { container, getByText } = render(
      <TestWrapper>
        <BaseScriptWithRef
          initialValues={initialValuesWithExpressions}
          allowableTypes={[MultiTypeInputType.EXPRESSION]}
        />
      </TestWrapper>
    )
    await waitFor(() => getByText('<+spec.environmentVariable.var1>'))
    expect(container).toMatchSnapshot()
  })

  test('should match snapshot for BaseScriptWithRef with initial script values with runtimeInput', async () => {
    const { container } = render(
      <TestWrapper>
        <BaseScriptWithRef initialValues={initialValuesWithRI} allowableTypes={[MultiTypeInputType.RUNTIME]} />
      </TestWrapper>
    )
    const configureBtn = container.querySelector('button[id="configureOptions_spec.source.spec.script"]')

    await waitFor(() => expect(configureBtn).toBeDefined())

    expect(container).toMatchSnapshot()
  })

  test('should match snapshot for BaseScriptWithRef without initial values ', async () => {
    const onChangeMock = jest.fn()
    const { container } = render(
      <TestWrapper>
        <Formik<any> formName="test-form" initialValues={defaultInitialValuesCorrect} onSubmit={jest.fn()}>
          {formik => {
            return (
              <Form>
                <BaseScriptWithRef
                  initialValues={defaultInitialValuesCorrect}
                  allowableTypes={[]}
                  ref={formik as any}
                  onChange={onChangeMock}
                />
              </Form>
            )
          }}
        </Formik>
      </TestWrapper>
    )

    const shell = container.querySelector('input[name="spec.shell"]')
    await waitFor(() => expect(shell).toBeDefined())

    act(() => {
      fillAtForm([
        {
          container,
          fieldId: 'spec.source.spec.script',
          type: InputTypes.TEXTAREA,
          value: 'echo test'
        }
      ])
    })

    expect(onChangeMock).toBeCalledWith({
      identifier: 'id',
      name: 'name',
      type: '',
      spec: {
        executionTarget: {},
        onDelegate: 'targethost',
        shell: 'Bash',
        source: {
          spec: {
            type: 'Inline',
            script: 'echo test'
          }
        }
      }
    })
  })
})
