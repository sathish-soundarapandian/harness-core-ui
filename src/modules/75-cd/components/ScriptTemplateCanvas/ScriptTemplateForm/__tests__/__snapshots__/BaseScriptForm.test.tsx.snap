// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test BaseScriptWithRef initial render 1`] = `
<div>
  <div
    class="OverlaySpinner--overlaySpinner"
  >
    <div
      class="formGroup sm"
    >
      <div
        class="bp3-form-group"
      >
        <label
          class="bp3-label"
          for="spec.shell"
        >
          <span
            class="Tooltip--acenter"
            data-tooltip-id="shellScriptForm_spec.shell"
          >
            common.scriptType
          </span>
           
          <span
            class="bp3-text-muted"
          />
        </label>
        <div
          class="bp3-form-content"
        >
          <div
            class="bp3-popover-wrapper Select--main bp3-fill"
          >
            <div
              class="bp3-popover-target"
            >
              <div
                class="bp3-input-group"
              >
                <input
                  autocomplete="off"
                  class="bp3-input"
                  name="spec.shell"
                  placeholder="- common.scriptType -"
                  style="padding-right: 0px;"
                  type="text"
                  value=""
                />
                <span
                  class="bp3-input-action"
                >
                  <span
                    class="bp3-icon bp3-icon-chevron-down StyledProps--main StyledProps--padding-small"
                    icon="chevron-down"
                  >
                    <svg
                      data-icon="chevron-down"
                      height="14"
                      viewBox="0 0 16 16"
                      width="14"
                    >
                      <desc>
                        chevron-down
                      </desc>
                      <path
                        d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="formGroup"
    >
      <div
        class="bp3-form-group"
      >
        <label
          class="bp3-label"
          for="spec.source.spec.script"
        >
          <div
            class="formLabel"
          >
            <span
              class="Tooltip--acenter"
              data-tooltip-id="shellScriptForm_spec.source.spec.script"
            >
              common.script
            </span>
            <div
              class="bp3-popover-wrapper typeSelectorWrapper"
            >
              <div
                class="bp3-popover-target typeSelector"
              >
                <button
                  class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main btn Button--iconOnly Button--without-shadow"
                  type="button"
                >
                  <span
                    class="bp3-button-text"
                  >
                    <span
                      class="bp3-icon StyledProps--main icon fixed"
                      data-icon="fixed-input"
                    >
                      <svg
                        height="12"
                        viewBox="0 0 9 9"
                        width="12"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M5.61 7.55c.62-.7.3-1.42.18-2.35a.32.32 0 01.1-.29l1.43-1.42a.76.76 0 01.28-.15c.39-.1 1 0 1.28-.29a.34.34 0 000-.51L6.46.12a.33.33 0 00-.52 0c-.28.29-.2.9-.28 1.3a.53.53 0 01-.13.23L4.09 3.12a.29.29 0 01-.28.09c-.89-.11-1.6-.49-2.27.11-.29.26-.29.43 0 .7l1.31 1.34.15.16-.15.15-2.7 2.69a.36.36 0 000 .54.36.36 0 00.54-.05l2.63-2.68.17-.17.17.16c.45.46.9.91 1.36 1.36.21.21.42.22.59.03z"
                          fill="currentColor"
                        />
                      </svg>
                    </span>
                  </span>
                </button>
              </div>
            </div>
          </div>
           
          <span
            class="bp3-text-muted"
          />
        </label>
        <div
          class="bp3-form-content"
        >
          <div
            class="monacoWrapper"
          >
            <textarea
              name="spec.source.spec.script"
            />
            <button
              class="bp3-button bp3-small Button--button StyledProps--font StyledProps--main expandBtn Button--with-current-color Button--iconOnly Button--withLeftIcon"
              type="button"
            >
              <span
                class="bp3-icon bp3-icon-fullscreen StyledProps--main"
                icon="fullscreen"
              >
                <svg
                  data-icon="fullscreen"
                  height="10"
                  viewBox="0 0 16 16"
                  width="10"
                >
                  <desc>
                    fullscreen
                  </desc>
                  <path
                    d="M3.41 2H5c.55 0 1-.45 1-1s-.45-1-1-1H1C.45 0 0 .45 0 1v4c0 .55.45 1 1 1s1-.45 1-1V3.41L5.29 6.7c.18.19.43.3.71.3a1.003 1.003 0 00.71-1.71L3.41 2zM6 9c-.28 0-.53.11-.71.29L2 12.59V11c0-.55-.45-1-1-1s-1 .45-1 1v4c0 .55.45 1 1 1h4c.55 0 1-.45 1-1s-.45-1-1-1H3.41l3.29-3.29c.19-.18.3-.43.3-.71 0-.55-.45-1-1-1zm9 1c-.55 0-1 .45-1 1v1.59L10.71 9.3A.965.965 0 0010 9a1.003 1.003 0 00-.71 1.71l3.3 3.29H11c-.55 0-1 .45-1 1s.45 1 1 1h4c.55 0 1-.45 1-1v-4c0-.55-.45-1-1-1zm0-10h-4c-.55 0-1 .45-1 1s.45 1 1 1h1.59l-3.3 3.29a1.003 1.003 0 001.42 1.42L14 3.41V5c0 .55.45 1 1 1s1-.45 1-1V1c0-.55-.45-1-1-1z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test BaseScriptWithRef should match snapshot for BaseScriptWithRef with initial script values with expressions 1`] = `
<div>
  <div
    class="OverlaySpinner--overlaySpinner"
  >
    <div
      class="formGroup sm"
    >
      <div
        class="bp3-form-group"
      >
        <label
          class="bp3-label"
          for="spec.shell"
        >
          <span
            class="Tooltip--acenter"
            data-tooltip-id="shellScriptForm_spec.shell"
          >
            common.scriptType
          </span>
           
          <span
            class="bp3-text-muted"
          />
        </label>
        <div
          class="bp3-form-content"
        >
          <div
            class="bp3-popover-wrapper Select--main bp3-fill"
          >
            <div
              class="bp3-popover-target"
            >
              <div
                class="bp3-input-group"
              >
                <input
                  autocomplete="off"
                  class="bp3-input"
                  name="spec.shell"
                  placeholder="- common.scriptType -"
                  style="padding-right: 0px;"
                  type="text"
                  value="Bash"
                />
                <span
                  class="bp3-input-action"
                >
                  <span
                    class="bp3-icon bp3-icon-chevron-down StyledProps--main StyledProps--padding-small"
                    icon="chevron-down"
                  >
                    <svg
                      data-icon="chevron-down"
                      height="14"
                      viewBox="0 0 16 16"
                      width="14"
                    >
                      <desc>
                        chevron-down
                      </desc>
                      <path
                        d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="formGroup"
    >
      <div
        class="bp3-form-group"
      >
        <label
          class="bp3-label"
          for="spec.source.spec.script"
        >
          <div
            class="formLabel"
          >
            <span
              class="Tooltip--acenter"
              data-tooltip-id="shellScriptForm_spec.source.spec.script"
            >
              common.script
            </span>
            <div
              class="bp3-popover-wrapper typeSelectorWrapper"
            >
              <div
                class="bp3-popover-target typeSelector"
              >
                <button
                  class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main btn Button--iconOnly Button--without-shadow"
                  type="button"
                >
                  <span
                    class="bp3-button-text"
                  >
                    <span
                      class="bp3-icon StyledProps--main icon expression"
                      data-icon="expression-input"
                    >
                      <svg
                        height="16"
                        viewBox="0 0 16 16"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          clip-rule="evenodd"
                          d="M4.875 2.588C5.305 2.218 5.889 2 6.647 2a.5.5 0 110 1c-.556 0-.898.155-1.12.346-.228.197-.374.472-.462.784a3.18 3.18 0 00-.106.86h1.449a.5.5 0 010 1H4.992v5.505a.5.5 0 01-1 0V5.99h-.936a.5.5 0 010-1h.903a4.173 4.173 0 01.143-1.13c.123-.435.35-.907.773-1.27zm.111 2.823c0 .001 0 .001 0 0m0 0zm7.059 5.019c-.242.268-.455.59-.553.786-.17.341-.472.705-.766.994-.285.281-.635.565-.932.684a.5.5 0 11-.371-.929c.102-.04.335-.204.6-.467.258-.253.475-.528.575-.729.142-.283.407-.68.704-1.009.15-.165.32-.33.5-.456.168-.118.406-.248.68-.248a.5.5 0 01.013 1c-.004.002-.042.012-.118.066a2.152 2.152 0 00-.332.307z"
                          fill="#CurrentColor"
                          fill-rule="evenodd"
                        />
                        <path
                          clip-rule="evenodd"
                          d="M10.045 10.43c.242.268.455.59.553.786.17.341.473.705.766.994.285.281.635.565.932.684a.5.5 0 10.372-.929c-.103-.04-.335-.204-.602-.467-.257-.253-.473-.528-.574-.729a5.003 5.003 0 00-.704-1.009 3.08 3.08 0 00-.5-.456c-.168-.118-.406-.248-.68-.248a.5.5 0 00-.012 1c.003.002.041.012.118.066a2.1 2.1 0 01.33.307z"
                          fill="#CurrentColor"
                          fill-rule="evenodd"
                        />
                        <path
                          clip-rule="evenodd"
                          d="M8.367 7.805a.5.5 0 01.186.682c-.854 1.494-.652 3.467.013 4.797a.5.5 0 11-.895.447c-.771-1.543-1.049-3.881.013-5.74a.5.5 0 01.683-.186zm5.273 0a.5.5 0 01.683.186c1.062 1.859.784 4.197.013 5.74a.5.5 0 11-.895-.447c.665-1.33.867-3.303.013-4.797a.5.5 0 01.186-.682z"
                          fill="currentColor"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </span>
                  </span>
                </button>
              </div>
            </div>
          </div>
           
          <span
            class="bp3-text-muted"
          />
        </label>
        <div
          class="bp3-form-content"
        >
          <div
            class="monacoWrapper"
          >
            <textarea
              name="spec.source.spec.script"
            >
              &lt;+spec.environmentVariable.var1&gt;
            </textarea>
            <button
              class="bp3-button bp3-small Button--button StyledProps--font StyledProps--main expandBtn Button--with-current-color Button--iconOnly Button--withLeftIcon"
              type="button"
            >
              <span
                class="bp3-icon bp3-icon-fullscreen StyledProps--main"
                icon="fullscreen"
              >
                <svg
                  data-icon="fullscreen"
                  height="10"
                  viewBox="0 0 16 16"
                  width="10"
                >
                  <desc>
                    fullscreen
                  </desc>
                  <path
                    d="M3.41 2H5c.55 0 1-.45 1-1s-.45-1-1-1H1C.45 0 0 .45 0 1v4c0 .55.45 1 1 1s1-.45 1-1V3.41L5.29 6.7c.18.19.43.3.71.3a1.003 1.003 0 00.71-1.71L3.41 2zM6 9c-.28 0-.53.11-.71.29L2 12.59V11c0-.55-.45-1-1-1s-1 .45-1 1v4c0 .55.45 1 1 1h4c.55 0 1-.45 1-1s-.45-1-1-1H3.41l3.29-3.29c.19-.18.3-.43.3-.71 0-.55-.45-1-1-1zm9 1c-.55 0-1 .45-1 1v1.59L10.71 9.3A.965.965 0 0010 9a1.003 1.003 0 00-.71 1.71l3.3 3.29H11c-.55 0-1 .45-1 1s.45 1 1 1h4c.55 0 1-.45 1-1v-4c0-.55-.45-1-1-1zm0-10h-4c-.55 0-1 .45-1 1s.45 1 1 1h1.59l-3.3 3.29a1.003 1.003 0 001.42 1.42L14 3.41V5c0 .55.45 1 1 1s1-.45 1-1V1c0-.55-.45-1-1-1z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test BaseScriptWithRef should match snapshot for BaseScriptWithRef with initial script values with runtimeInput 1`] = `
<div>
  <div
    class="OverlaySpinner--overlaySpinner"
  >
    <div
      class="formGroup sm"
    >
      <div
        class="bp3-form-group"
      >
        <label
          class="bp3-label"
          for="spec.shell"
        >
          <span
            class="Tooltip--acenter"
            data-tooltip-id="shellScriptForm_spec.shell"
          >
            common.scriptType
          </span>
           
          <span
            class="bp3-text-muted"
          />
        </label>
        <div
          class="bp3-form-content"
        >
          <div
            class="bp3-popover-wrapper Select--main bp3-fill"
          >
            <div
              class="bp3-popover-target"
            >
              <div
                class="bp3-input-group"
              >
                <input
                  autocomplete="off"
                  class="bp3-input"
                  name="spec.shell"
                  placeholder="- common.scriptType -"
                  style="padding-right: 0px;"
                  type="text"
                  value="Bash"
                />
                <span
                  class="bp3-input-action"
                >
                  <span
                    class="bp3-icon bp3-icon-chevron-down StyledProps--main StyledProps--padding-small"
                    icon="chevron-down"
                  >
                    <svg
                      data-icon="chevron-down"
                      height="14"
                      viewBox="0 0 16 16"
                      width="14"
                    >
                      <desc>
                        chevron-down
                      </desc>
                      <path
                        d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="formGroup"
    >
      <div
        class="bp3-form-group formGroup"
      >
        <label
          class="bp3-label"
          for="spec.source.spec.script"
        >
          <div
            class="formLabel"
          >
            <span
              class="Tooltip--acenter"
              data-tooltip-id="shellScriptForm_spec.source.spec.script"
            >
              common.script
            </span>
            <div
              class="bp3-popover-wrapper typeSelectorWrapper"
            >
              <div
                class="bp3-popover-target typeSelector"
              >
                <button
                  class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main btn Button--iconOnly Button--without-shadow"
                  type="button"
                >
                  <span
                    class="bp3-button-text"
                  >
                    <span
                      class="bp3-icon StyledProps--main icon runtime"
                      data-icon="runtime-input"
                    >
                      <svg
                        height="12"
                        viewBox="0 0 10 7.97"
                        width="12"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M8.14 4L8 3.72a2.53 2.53 0 01-.12-.3l-.16-.35c-.11-.25-.22-.38-.32-.38s-.28.15-.37.44C7 3.1 7 3.07 7 3a.73.73 0 01.24-.46.66.66 0 01.42-.26c.25 0 .5.35.76 1.05l.07.17.07-.12c.47-.73.86-1.1 1.18-1.1a.71.71 0 01.29.07l-.4.42h-.11c-.24 0-.52.28-.86.83l-.09.16.07.2c.29.79.53 1.18.71 1.18s.31-.12.4-.36q.09.06.09.12t-.27.39a.74.74 0 01-.45.23c-.25 0-.51-.36-.77-1.07l-.12-.25-.1.17c-.44.76-.87 1.14-1.28 1.14a.65.65 0 01-.39-.12L6.84 5a.32.32 0 00.23.1c.25 0 .53-.25.83-.74l.17-.28zm-4-.32L1.7.59h2.47S5 .52 5.35 1a3.28 3.28 0 01.42 1.14h.31L5.94 0H0v.31l3.06 3.74L0 7.62V8h6.08l.38-2.36-.31-.07a2.43 2.43 0 01-.52 1c-.42.48-2 .41-2 .41H1.39z"
                          fill="current-color"
                        />
                      </svg>
                    </span>
                  </span>
                </button>
              </div>
            </div>
          </div>
           
          <span
            class="bp3-text-muted"
          />
        </label>
        <div
          class="bp3-form-content"
        >
          <div
            class="bp3-form-group bp3-disabled runtimeDisabled"
          >
            
            <div
              class="bp3-form-content"
            >
              <div
                class="bp3-input-group bp3-disabled"
              >
                <input
                  autocomplete="off"
                  class="bp3-input"
                  disabled=""
                  name="spec.source.spec.script"
                  type="text"
                  value="<+input>"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <span
        class="bp3-popover-wrapper bp3-dark"
      >
        <span
          class="bp3-popover-target"
        >
          <button
            class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main minConfigBtn Button--with-current-color Button--iconOnly Button--without-shadow Button--withRightIcon"
            id="configureOptions_spec.source.spec.script"
            tabindex="0"
            type="button"
          >
            <span
              class="bp3-icon bp3-icon-cog StyledProps--main"
              icon="cog"
            >
              <svg
                data-icon="cog"
                height="16"
                viewBox="0 0 16 16"
                width="16"
              >
                <desc>
                  cog
                </desc>
                <path
                  d="M15.19 6.39h-1.85c-.11-.37-.27-.71-.45-1.04l1.36-1.36c.31-.31.31-.82 0-1.13l-1.13-1.13a.803.803 0 00-1.13 0l-1.36 1.36c-.33-.17-.67-.33-1.04-.44V.79c0-.44-.36-.8-.8-.8h-1.6c-.44 0-.8.36-.8.8v1.86c-.39.12-.75.28-1.1.47l-1.3-1.3c-.3-.3-.79-.3-1.09 0L1.82 2.91c-.3.3-.3.79 0 1.09l1.3 1.3c-.2.34-.36.7-.48 1.09H.79c-.44 0-.8.36-.8.8v1.6c0 .44.36.8.8.8h1.85c.11.37.27.71.45 1.04l-1.36 1.36c-.31.31-.31.82 0 1.13l1.13 1.13c.31.31.82.31 1.13 0l1.36-1.36c.33.18.67.33 1.04.44v1.86c0 .44.36.8.8.8h1.6c.44 0 .8-.36.8-.8v-1.86c.39-.12.75-.28 1.1-.47l1.3 1.3c.3.3.79.3 1.09 0l1.09-1.09c.3-.3.3-.79 0-1.09l-1.3-1.3c.19-.35.36-.71.48-1.1h1.85c.44 0 .8-.36.8-.8v-1.6a.816.816 0 00-.81-.79zm-7.2 4.6c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </button>
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`Test BaseScriptWithRef should match snapshot for BaseScriptWithRef with initial values 1`] = `
<div>
  <div
    class="OverlaySpinner--overlaySpinner"
  >
    <div
      class="formGroup sm"
    >
      <div
        class="bp3-form-group"
      >
        <label
          class="bp3-label"
          for="spec.shell"
        >
          <span
            class="Tooltip--acenter"
            data-tooltip-id="shellScriptForm_spec.shell"
          >
            common.scriptType
          </span>
           
          <span
            class="bp3-text-muted"
          />
        </label>
        <div
          class="bp3-form-content"
        >
          <div
            class="bp3-popover-wrapper Select--main bp3-fill"
          >
            <div
              class="bp3-popover-target"
            >
              <div
                class="bp3-input-group"
              >
                <input
                  autocomplete="off"
                  class="bp3-input"
                  name="spec.shell"
                  placeholder="- common.scriptType -"
                  style="padding-right: 0px;"
                  type="text"
                  value="Bash"
                />
                <span
                  class="bp3-input-action"
                >
                  <span
                    class="bp3-icon bp3-icon-chevron-down StyledProps--main StyledProps--padding-small"
                    icon="chevron-down"
                  >
                    <svg
                      data-icon="chevron-down"
                      height="14"
                      viewBox="0 0 16 16"
                      width="14"
                    >
                      <desc>
                        chevron-down
                      </desc>
                      <path
                        d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="formGroup"
    >
      <div
        class="bp3-form-group"
      >
        <label
          class="bp3-label"
          for="spec.source.spec.script"
        >
          <div
            class="formLabel"
          >
            <span
              class="Tooltip--acenter"
              data-tooltip-id="shellScriptForm_spec.source.spec.script"
            >
              common.script
            </span>
            <div
              class="bp3-popover-wrapper typeSelectorWrapper"
            >
              <div
                class="bp3-popover-target typeSelector"
              >
                <button
                  class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main btn Button--iconOnly Button--without-shadow"
                  type="button"
                >
                  <span
                    class="bp3-button-text"
                  >
                    <span
                      class="bp3-icon StyledProps--main icon fixed"
                      data-icon="fixed-input"
                    >
                      <svg
                        height="12"
                        viewBox="0 0 9 9"
                        width="12"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M5.61 7.55c.62-.7.3-1.42.18-2.35a.32.32 0 01.1-.29l1.43-1.42a.76.76 0 01.28-.15c.39-.1 1 0 1.28-.29a.34.34 0 000-.51L6.46.12a.33.33 0 00-.52 0c-.28.29-.2.9-.28 1.3a.53.53 0 01-.13.23L4.09 3.12a.29.29 0 01-.28.09c-.89-.11-1.6-.49-2.27.11-.29.26-.29.43 0 .7l1.31 1.34.15.16-.15.15-2.7 2.69a.36.36 0 000 .54.36.36 0 00.54-.05l2.63-2.68.17-.17.17.16c.45.46.9.91 1.36 1.36.21.21.42.22.59.03z"
                          fill="currentColor"
                        />
                      </svg>
                    </span>
                  </span>
                </button>
              </div>
            </div>
          </div>
           
          <span
            class="bp3-text-muted"
          />
        </label>
        <div
          class="bp3-form-content"
        >
          <div
            class="monacoWrapper"
          >
            <textarea
              name="spec.source.spec.script"
            >
              echo test
            </textarea>
            <button
              class="bp3-button bp3-small Button--button StyledProps--font StyledProps--main expandBtn Button--with-current-color Button--iconOnly Button--withLeftIcon"
              type="button"
            >
              <span
                class="bp3-icon bp3-icon-fullscreen StyledProps--main"
                icon="fullscreen"
              >
                <svg
                  data-icon="fullscreen"
                  height="10"
                  viewBox="0 0 16 16"
                  width="10"
                >
                  <desc>
                    fullscreen
                  </desc>
                  <path
                    d="M3.41 2H5c.55 0 1-.45 1-1s-.45-1-1-1H1C.45 0 0 .45 0 1v4c0 .55.45 1 1 1s1-.45 1-1V3.41L5.29 6.7c.18.19.43.3.71.3a1.003 1.003 0 00.71-1.71L3.41 2zM6 9c-.28 0-.53.11-.71.29L2 12.59V11c0-.55-.45-1-1-1s-1 .45-1 1v4c0 .55.45 1 1 1h4c.55 0 1-.45 1-1s-.45-1-1-1H3.41l3.29-3.29c.19-.18.3-.43.3-.71 0-.55-.45-1-1-1zm9 1c-.55 0-1 .45-1 1v1.59L10.71 9.3A.965.965 0 0010 9a1.003 1.003 0 00-.71 1.71l3.3 3.29H11c-.55 0-1 .45-1 1s.45 1 1 1h4c.55 0 1-.45 1-1v-4c0-.55-.45-1-1-1zm0-10h-4c-.55 0-1 .45-1 1s.45 1 1 1h1.59l-3.3 3.29a1.003 1.003 0 001.42 1.42L14 3.41V5c0 .55.45 1 1 1s1-.45 1-1V1c0-.55-.45-1-1-1z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
